controller_manager/controller_manager/spawner.py

```diff
@@ -135,7 +135,7 @@ def is_controller_loaded(node, controller_manager, controller_name):
 def main(args=None):
     rclpy.init(args=args, signal_handler_options=SignalHandlerOptions.NO)
     parser = argparse.ArgumentParser()
-    parser.add_argument("controller_name", help="Name of the controller")
```

Comment:
I'd like to merge this into Iron and maybe even Humble but this would break every launch file using the script.

Could you please keep `controller_name` as a declared argument in argparse and post some warning when it is used (but otherwise just feed it into `args.controller_names = [args.controller_name]` so the rest of the code can be updated as you've done already

Something like this? https://gist.github.com/bsolomon1124/44f77ed2f15062c614ef6e102bc683a5
Looks fairly elegant to me

Reply:
I think you are correct! Thanks for looking into it